### PR TITLE
New license key obfuscation algorithm

### LINKED ISSUE #2861: License key redaction - Leave the first 10 characters visible
### Description
GTS has requested that when Agents log license keys, rather than redacting the whole thing, leave the first 10 characters un-redacted. This will make it easier to troubleshoot certain kinds of customer issues, including when the wrong region is being used.

From the spec update:

Agents **MUST NOT** log a license key in full. When a license key appears in log output (debug logs, audit logs, error messages, etc.), the agent **MUST** redact it such that the first 10 characters are visible with the remaining characters replaced by `*` as the redaction marker. Keys **MUST** retain their original length after redaction.

If a key is 10 or fewer characters (e.g. a short invalid key), the agent **SHOULD** fully redact it.

Internal [spec](https://source.datanerd.us/agents/agent-specs/pull/793) change

### PR DESCRIPTION
**Pull request by jtduffy**
Resolves #2861 , #2813 

Updates the license key obfuscation rules based on the updated spec.

Addresses edge case where license key can be exposed if it's changed while the agent is running.

Keys will now show up as follows with `audit_mode` on:

Connect: `https://staging-collector.newrelic.com:443/agent_listener/invoke_raw_method?method=preconnect&license_key=0000000000******************************&marshal_format=json&protocol_version=17`

Config payload: `"license_key":"0000000000******************************"`

### CODE REVIEW

File: newrelic-agent/src/main/java/com/newrelic/agent/util/LicenseKeyUtil.java [hunk 1]
```diff
@@ -32,7 +37,40 @@ public static String obfuscateLicenseKey(String originalString) {
             Agent.LOG.finest("Unable to obfuscate a null license_key.");
             return originalString;
```
Comment (kanderson250) [inline newrelic-agent/src/main/java/com/newrelic/agent/util/LicenseKeyUtil.java:43]: In the extremely weird edge case the user deletes their license key from config after startup, the original key is still logged.

File: newrelic-agent/src/test/java/com/newrelic/agent/util/LicenseKeyUtilTest.java [hunk 2]
```diff
@@ -116,4 +116,59 @@ public void testObfuscateLicenseKeyWithNullOrEmptyString() {
...
+        AgentConfig config = AgentConfigImpl.createAgentConfig(configMap);
+        ConfigService configService = ConfigServiceFactory.createConfigService(config, configMap);
+        serviceManager.setConfigService(configService);
+
+        // When
+        String actualRequestUrl = LicenseKeyUtil.obfuscateLicenseKey(originalRequestUrl);
+        String actualJsonPayload = LicenseKeyUtil.obfuscateLicenseKey(originalJsonPayload);
+
+        // Then
+        String expectedRequestUrl = "https://staging-collector.newrelic.com:443/agent_listener/invoke_raw_method?method=connect&license_key=this_is_a_*************************&marshal_format=json&protocol_version=17";
```
Comment (kanderson250) [inline newrelic-agent/src/test/java/com/newrelic/agent/util/LicenseKeyUtilTest.java]: Left a comment higher originally but this test is a much better place!

I don't think this is what we want the expected behavior to be - shouldn't we be obfuscating the original request url that's passed in (since that is the data sender's actual url destination), not whatever is read from config?

If Data Sender sends in 

`https://staging-collector.newrelic.com:443/agent_listener/invoke_raw_method?method=connect&license_key=abcdefghijklmonpqrstuvwxyz1234567890&marshal_format=json&protocol_version=17` 

I would expect 

`https://staging-collector.newrelic.com:443/agent_listener/invoke_raw_method?method=connect&license_key=abcdefghij*****************&marshal_format=json&protocol_version=17` to come out.

Comment (jtduffy) [reply newrelic-agent/src/test/java/com/newrelic/agent/util/LicenseKeyUtilTest.java]: Yes, I corrected this when I messing with the AIT failures. It no longer relies on the key read from config

**Review by jbedell-newrelic — APPROVED** [review]
